packages/aws-cdk-lib/core/lib/cfn-fn.ts

```diff
@@ -334,6 +334,10 @@ export class Fn {
     if (conditions.length === 1) {
       return conditions[0] as ICfnRuleConditionExpression;
     }
+    // prevent the error "Fn::Or object requires a list of at least 2" when the condition is split into groups of 10 and 1
+    if (conditions.length % 10 === 1) {
+      return Fn.conditionOr(..._inGroupsOf(conditions, 9).map(group => new FnOr(...group)));
```

Comment:
I think this can break when `conditions.length % 90 == 1` :)

Reply:
Yes, but would it be necessary to consider conditions for numbers larger than 91 or 181...?

Reply:
I agree that if we can't be sure, I should respond.
But in this case, I don't have a simple solution to solve this 91 or 181 or more problem...
And I'd prefer not to add more complex patch and test(should check the case of 91 and doesn't use snapshot for condition now) for funky code of 91 or 181 or more conditions.

Reply:
I see. I feel the solution @MalikAtalla-AWS proposed is also simple without any edge cases, but let's wait for comments from the maintainers.